### ISSUE TITLE
Marking type period selects always same month

### ISSUE THREAD

**Issue by cruzms** [wix/react-native-calendars#1821]
## Description

When the `markingType` is set to `period` the onDayPress callback always return a day on the default month and automatically switches to default month even if no onDayPress is passed in properties

## Expected Behavior

The callback should return the current selected date, as when any other `markingType` is set

https://user-images.githubusercontent.com/31360984/157355609-7f10f40a-81e7-454c-8368-acab936976ea.mp4

## Observed Behavior

https://user-images.githubusercontent.com/31360984/157355592-8006a91b-7ff3-46b8-b939-e72b06ae685e.mp4

## Environment

* `npm ls react-native-calendars`:  react-native-calendars@1.1279.0
* `npm ls react-native`: react-native@0.64.3

Also specify:

1. Device/emulator/simulator & OS version:

Emulator: Pixel_2_API_29
Device: Redmi Note 7 Android 10

## Reproducible Demo

``` js
export default function App() {
  const [show, setShow] = useState(false);
  const [selectedDate, setSelectedDate] = useState("");
  return (
    <View>
      <Calendar
        onDayPress={(data) => {
          console.log(data);
          setSelectedDate(data.dateString);
        }}
        markingType="period"
      />
      <Text>{selectedDate}</Text>
    </View>
  );
}
```

**Comment by ilaloov** [comment]
I have the same issue:

- `react-native-calendars: 1.1279.0`
- `react-native: 0.64.3`

Testing on physical device with iOS 15.3.1

**Comment by eramudeep** [comment]
> I have the same issue with the most recent release (1.1279.0). The issue does not occur in 1.1278.0.
> 
> * `react-native-calendars: 1.1279.0`
> * `react-native: 0.64.3`
> 
> Testing on physical device with iOS 15.3.1

yes, i was also facing the same issue. then switch back to version **1.1276.0**
and start working.
seems like many things are broken in the latest version.
or the documentation is not upto date.

**Comment by bruno-fialho** [comment]
In my case, switching back to 1.1276.0 didn't resolve this issue. 

> react-native-calendars: 1.1279.0
react-native: 0.64.3

Same problem for device iPhone 12 (iOS 15.3.1) and emulator iPhone 13 (iOS 15.2)

Code:

```js
<CustomCalendar
    firstDay={1}
    minDate={String(new Date())}
    markingType="period"
    markedDates={markedDates}
    onDayPress={onDayPress}
/>
```

Screen:

https://user-images.githubusercontent.com/45835631/159137066-bd88b1a8-4c08-40a6-b3a0-1d3114aa0a2f.mov

**Comment by doougui** [comment]
Any updates on this?

**Comment by alevkov** [comment]
Is anyone looking into this? Seems like either a bug, or unintuitive and undocumented behavior. This is a blocker for our use case.

**Comment by alevkov** [comment]
cc: @ethanshar

**Comment by doougui** [comment]
> Is anyone looking into this? Seems like either a bug, or unintuitive and undocumented behavior. This is a blocker for our use case.

Downgrading to `1.1278.0` solved my problem (at least for now). It might work for you too. Unfortunately that seems like the only way to use this feature properly right now :(

**Comment by Inbal-Tish** [comment]
@cruzms Hi. Can you please check on Calendars version `1.1280.0` and tell me if the issue persists?

**Comment by cruzms** [comment]
> @cruzms Hi. Can you please check on Calendars version `1.1280.0` and tell me if the issue persists?

Hi, is working as expected in version `1.1280.0`

**Comment by Inbal-Tish** [comment]
@cruzms Great, so fix on version 1.1280.0. Closing the issue. Thank you for reporting!!
